FaunaDB.Client/Errors/FaunaException.cs

```diff
@@ -10,114 +8,158 @@ namespace FaunaDB.Errors
+    {
+        public ValidationFailed(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.ValidationFailed, message, position) { }
+    }
+
+    public class InstanceNotUnique : FaunaException
+    {
+        public InstanceNotUnique(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InstanceNotUnique, message, position) { }
+    }
+
+    public class FeatureNotAvailable : FaunaException
+    {
+        public FeatureNotAvailable(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.FeatureNotAvailable, message, position) { }
+    }
+
+    public class ValueNotFound : FaunaException
+    {
+        public ValueNotFound(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.ValueNotFound, message, position) { }
+    }
+
+    public class InstanceNotFound : FaunaException
+    {
+        public InstanceNotFound(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InstanceNotFound, message, position) { }
+    }
+
+    public class AuthenticationFailed : FaunaException
+    {
+        public AuthenticationFailed(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.AuthenticationFailed, message, position) { }
+    }
+
+    public class InvalidArgument : FaunaException
+    {
+        public InvalidArgument(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InvalidArgument, message, position) { }
+    }
+
+    public class TransactionAborted : FaunaException
+    {
+        public TransactionAborted(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.TransactionAborted, message, position) { }
+    }
+
+    public class InvalidWriteTime : FaunaException
+    {
+        public InvalidWriteTime(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InvalidWriteTime, message, position) { }
+    }
+
+    public class MissingIdentity : FaunaException
+    {
+        public MissingIdentity(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.MissingIdentity, message, position) { }
+    }
+
+    public class InvalidScope : FaunaException
+    {
+        public InvalidScope(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InvalidScope, message, position) { }
+    }
+
+    public class InvalidToken : FaunaException
+    {
+        public InvalidToken(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.InvalidToken, message, position) { }
+    }
+
+    public class CallError : FaunaException
+    {
+        public CallError(int httpStatusCode, string message, string[] position) : base(httpStatusCode, ExceptionCodes.CallError, message, position) { }
```

Comment:
There is also an internal `cause` object that needs to be added here, check out a Java driver for this:https://github.com/fauna/faunadb-jvm/blob/OSS-864/faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java#L536